opensrp-chw/src/main/java/org/smartregister/chw/activity/MalariaProfileActivity.java

```diff
@@ -0,0 +1,131 @@
+package org.smartregister.chw.activity;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Toast;
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.domain.Form;
+import org.json.JSONObject;
+import org.smartregister.chw.R;
+import org.smartregister.chw.malaria.activity.BaseMalariaProfileActivity;
+import org.smartregister.chw.malaria.domain.MemberObject;
+import org.smartregister.chw.malaria.util.Constants;
+import org.smartregister.commonregistry.CommonPersonObject;
+import org.smartregister.commonregistry.CommonPersonObjectClient;
+import org.smartregister.commonregistry.CommonRepository;
+import org.smartregister.family.util.JsonFormUtils;
+import org.smartregister.family.util.Utils;
+import timber.log.Timber;
+
+public class MalariaProfileActivity extends BaseMalariaProfileActivity {
+    private final String TAG = MalariaProfileActivity.class.getCanonicalName();
+    private static final String CLIENT = "client";
+    private View view;
+    private MemberObject memberObject;
+    private ChildProfileActivityFlv flavor = new ChildProfileActivityFlv();
+
+    public static void startMalariaActivity(Activity activity, MemberObject memberObject, CommonPersonObjectClient client) {
+        Intent intent = new Intent(activity, MalariaProfileActivity.class);
+        intent.putExtra(Constants.MALARIA_MEMBER_OBJECT.MEMBER_OBJECT, memberObject);
+        intent.putExtra(CLIENT, client);
+        activity.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreation() {
+        super.onCreation();
+        memberObject = (MemberObject) getIntent().getSerializableExtra(Constants.MALARIA_MEMBER_OBJECT.MEMBER_OBJECT);
+    }
+
+    @Override
+    public void onClick(View view) {
+        onBackPressed();
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.other_member_menu, menu);
+        if (flavor.showMalariaConfirmationMenu()) {
+            menu.findItem(R.id.action_malaria_registration).setVisible(false);
+            menu.findItem(R.id.action_malaria_followup_visit).setVisible(true);
+        }
+        menu.findItem(R.id.action_anc_registration).setVisible(false);
```

Comment:
Should we be doing ANC registration on the malaria profile page ?

Reply:
I think we may need to avoid adding views that we don't need just to set the visibility to false this will just bloat up the module

Reply:
should we be adding them programmatically?

Reply:
the menu is shared across multiple classes, its false because its not needed in this scenario. But in another class it could be vice versa

Reply:
Would it not be better to have a layout menu specific to malaria profile? using a shared one would mean if the other_member_menu adds a new item then you would need to also update the malaria profile activity